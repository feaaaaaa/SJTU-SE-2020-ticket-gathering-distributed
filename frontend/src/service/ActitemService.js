/**
 * @param actitemid {int}
 *
 */

import {authRequest} from "../utils/ajax";

export const getDetail = (value,token,callback) => {
    const data={actitemid:value};
    const url = `/Actitem/detail`;
    authRequest(url, data,token,callback);
};