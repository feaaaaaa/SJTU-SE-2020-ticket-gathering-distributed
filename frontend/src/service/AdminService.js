import {authRequest} from "../utils/ajax";

export const addActivity = (value,token,callback) => {
    const data={activity:value};
    const url = '/activity/add';
    authRequest(url, data,token, callback);
};
// export const addActivity = (id,value,token,callback) => {
//     const data={id:id,activity:value};
//     const url = '/activity/add';
//     authRequest(url, data,token, callback);
// };

export const deleteActivity=(value,token,callback)=>{
    const data={activityId:value};
    const url='/activity/delete';
    authRequest(url,data,token,callback);
}