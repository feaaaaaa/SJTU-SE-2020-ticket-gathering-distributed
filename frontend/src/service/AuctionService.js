import {authRequest, postRequest} from "../utils/ajax";

export const getAuctions = (token,callback) => {
    const data={searching:null};
    const url = `/auction/get`;
    authRequest(url, data,token, callback);
};

export const addAuction = (actitemid,ddl,showtime,initprice,orderprice,amount,token,callback) => {
    const data={actitemid:actitemid,ddl:ddl,showtime:showtime,initprice:initprice,orderprice:orderprice,amount:amount};
    const url = `/auction/add`;
    console.log(data);
    authRequest(url, data,token, callback);
};

export const joinAuctions = (auctionid,userid,price,token,callback) => {
    const data={auctionid:auctionid,userid:userid,price:price};
    const url = `/auction/join`;
    authRequest(url, data,token, callback);
};

export const enterAuction=(userId,auctionId,token,callback)=>{
    const data={userid:userId,auctionid:auctionId};
    const url = `/auction/canEnter`;
    authRequest(url,data,token,callback);
}

export const deposit=(userId,auctionId,token,callback)=>{
    const data={userid:userId,auctionid:auctionId};
    const url = `/auction/deposit`;
    authRequest(url,data,token,callback);
}

export const getPriceById=(auctionId,token,callback)=>{
    const data={auctionid:auctionId};
    const url = `/auction/getPrice`;
    authRequest(url,data,token,callback);
}
