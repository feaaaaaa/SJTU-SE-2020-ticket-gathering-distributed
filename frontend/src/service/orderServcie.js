/**
 * @param userId {number}
 *
 *
 */
import {authRequest} from "../utils/ajax";

export const getOrderInfoByUser=(value,token,callback)=>{
    const apiUrl ='/Order/GetOrderInfoByUser';
    const data={userId:value};
    authRequest(apiUrl,data,token,callback);
}

export const addOrder=(userId,actitemId,price,amount,showtime,orderTime,token,callback)=>
{
    const apiUrl='/Order/addOrder';
    const data={userId:userId,actitemId:actitemId,price:price,amount:amount,showtime:showtime,orderTime:orderTime};
    console.log(data);
    authRequest(apiUrl,data,token,callback);
}