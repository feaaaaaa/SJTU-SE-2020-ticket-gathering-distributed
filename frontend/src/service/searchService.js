import {postRequest,fetchPost1} from "../utils/ajax";

export const search = (value,page,callback) => {
    const data={search:value,page:page};
    const url = '/activity/search';
    postRequest(url, data, callback);
};

export const getSearchPageNum=(value,callback)=>{
    const data={search:value};
    const url='/activity/searchPageNum';
    postRequest(url,data,callback);
}

// export const category_search = (value,callback) => {
//     const data = value;
//     const url = '/Activity/FindActivityByCategory';
//     console.log(value);
//     fetchPost1(url,data,callback);
// }

export const getSelectSearchPageNum=(type,name,city,callback)=>{
    const data={type:type,name:name,city:city};
    const url='/activity/FindActivityByCategoryPageNum';
    postRequest(url,data,callback);
}

export const category_search = (type,name,city,page,callback) => {
    console.log(type);
    console.log(name);
    console.log(city);
    const data={type:type,name:name,city:city,page:page};
    const url = '/activity/FindActivityByCategory';
    postRequest(url,data,callback);
}

export const homeSearch=(callback)=>{
    const url = '/activity/FindActivityByCategoryHome';
    const data={};
    postRequest(url,data,callback);
}
