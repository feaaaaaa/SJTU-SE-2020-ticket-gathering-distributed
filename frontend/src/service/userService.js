import {postRequest,fetchPost1} from "../utils/ajax";

export const checkUser = (value,callback) => {
    const data={username:value};
    const url = `http://localhost:8080/ExistsByUsername`;
    postRequest(url, data, callback);
};


export const login = (value,callback) => {
    const data={username:value.username,password:value.password}
    const url = 'http://localhost:8080/Login';
    fetchPost1(url, data, callback);
};

export const register = (username,password,email,personicon,phone,gender,callback) => {
    const url = 'http://localhost:8080/Register';
    const data={
        username:username,
        password:password,
        type:"User",
        gender: gender,
        email:email,
        phone:phone,
        personIcon:personicon
    };
    postRequest(url, data, callback);
};

// export const register = (self) => {
//     const url = 'http://localhost:8080/User/Register';
//     const callback = (data) => {
//         if(data.status >= 0) {
//             localStorage.setItem('userId', data.data.userId);
//             localStorage.setItem('userName', data.data.username);
//             self.$message({
//                 message: "Register successfully",//data.msg,
//                 type: "success",
//             });
//             self.$router.replace("/");
//         }
//         else{
//             //message.error(data.msg);
//             self.$message({
//                 message: "Username is repetitive! Please choose another one.",//data.msg,
//                 type: "error",
//             });
//         }
//     };
//     let user={username:self.username,password:self.password,type:"User",gender: "Male",email:self.email,phone:self.phone,personIcon:image2Base64(self.personicon)};
//     postRequest(url, user, callback);
// };

// function image2Base64(img) {
//     var canvas = document.createElement("canvas");
//     canvas.width = img.width;
//     canvas.height = img.height;
//     var ctx = canvas.getContext("2d");
//     ctx.drawImage(img, 0, 0, img.width, img.height);
//     var dataURL = canvas.toDataURL("image/png");
//     return dataURL;
// }