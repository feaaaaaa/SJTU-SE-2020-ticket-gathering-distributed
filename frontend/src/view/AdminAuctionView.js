import React from 'react';
import {HeaderInfo} from "../component/Header";
import {FooterInfo} from "../component/Footer";
import {AdminAuction} from "../component/AdminAuction";
import {Redirect} from "react-router-dom";
import "../css/Detail.css";
import {RecommendList} from "../component/RecommendList";


export class AdminAuctionView extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            isSearch:false,
            search:null
        };
        this.onSearch=this.onSearch.bind(this);
    }

    onSearch(value){
        this.setState({isSearch:true,search:value});
    }

    render() {
        if(this.state.isSearch){
            console.log("jumping...");
            return <Redirect to={{
                pathname: "/sortPage",
                state:{
                    search:this.state.search
                }
            }}/>;
        }
        return (
            <div>
                <HeaderInfo search={this.onSearch}/>
                <div id="Detail" style={{paddingTop:150,float:"left",marginLeft:-100,width:"80%"}}>
                    <AdminAuction />
                </div>
                <div style={{paddingTop:1000}}>
                    <FooterInfo />
                </div>
            </div>
        );
    }
}