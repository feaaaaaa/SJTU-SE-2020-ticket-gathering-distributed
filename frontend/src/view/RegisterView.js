import React from 'react';
import '../App.css'
import '../css/register.css'
import {Button, Input, Layout, message, Alert, Result, Card,Radio} from 'antd';
import {  UserOutlined } from '@ant-design/icons';
import {findUser, addUser, register} from "../service/userService";
import {UploadImage} from "../component/UploadImage";
import {Link}  from 'react-router-dom';

/**
 * @ClassName RegisterView
 * @Description register page
 * @Author Yang Yicheng
 * @Date 2020/7/10
 */

export class RegisterView extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            username:null,
            password:null,
            email:null,
            phone:null,
            personicon:null,
            gender:0,
            usernameErrorVisible:false,
            passwordErrorVisible:false,
            emailErrorVisible:false,
            registered:false,
        };
        this.handleUsername=this.handleUsername.bind(this);
        this.handlePassword=this.handlePassword.bind(this);
        this.checkPassword=this.checkPassword.bind(this);
        this.handleEmail=this.handleEmail.bind(this);
        this.handlePhone=this.handlePhone.bind(this);
        this.handleCommitImage=this.handleCommitImage.bind(this);
        this.handleGender=this.handleGender.bind(this);
    }

    handleUsername(e){
        const value=e.target.value;
        // const callback=(data)=>{
        //     if(data){
        //         this.setState({usernameErrorVisible:true});
        //     }
        //     else{
        //         this.setState({username:value,usernameErrorVisible:false})
        //     }
        // }
        // findUser(value,callback);

        /**frontend only*/
        this.setState({username:value});
    }

    handlePassword(e){
        this.setState({password:e.target.value})
    }

    checkPassword(e){
        const value=e.target.value;
        if(value==""){
            this.setState({password:null});
        }
        else{
            if (value!=this.state.password){
                this.setState({passwordErrorVisible:true})
            }
            else{
                this.setState({passwordErrorVisible:false})
            }
        }
    }

    handlePhone(e){
        const value=e.target.value;
        this.setState({phone:value});
    }

    handleEmail(e){
        let value = e.target.value;
        console.log(value);
        if(value==""){
            this.setState({email:null});
        }
        else{
            if (!(/^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/.test(value))) {
                this.setState({emailErrorVisible: true});
            } else {
                this.setState({email: value, emailErrorVisible: false});
            }
        }
    }

    handleSubmit(){
        // const callback =  (data) => {
        //     this.setState({registered:true});
        // };
        // if(this.state.username!=null &&this.state.password!=null &&this.state.email!=null){
        //     if(!this.state.usernameErrorVisible && !this.state.passwordErrorVisible && !this.state.emailErrorVisible){
        //         addUser(this.state.username,this.state.password,this.state.address,callback);
        //     }
        // }

        /**frontend only*/
        // this.setState({registered:true});
        register(this.state.username,this.state.password,this.state.address,self);
    }

    handleCommitImage(data){
        this.setState({personicon:data});
    }

    handleGender(e){
        this.setState({gender:e.target.value});
    }
    render(){
        if(this.state.registered){
            return(
                <div id="bg">
                <div id="regDiv">
                <Result
                    status="success"
                    title="注册成功"
                    subTitle="欢迎来到聚票网"
                    extra={[
                        <Link to={{
                            pathname: '/login',
                        }}
                        >
                            <Button type="primary" key="console">
                                去登陆
                            </Button>
                        </Link>
                    ]}
                />
                </div>
                </div>
            )
        }
        return(
            <div id="bg">
            <div id="regDiv">
                <p>请输入用户名:</p>
                <Input
                    placeholder="Enter your username"
                    prefix={<UserOutlined className="site-form-item-icon" />}
                    onBlur={this.handleUsername}
                >
                </Input>
                <div>
                    {this.state.usernameErrorVisible?(
                        <Alert message="该用户名已被注册过" type="error"/>
                    ):null}
                </div>
                <p>请输入密码:</p>
                <Input.Password
                    placeholder="Enter your password"
                    prefix={<UserOutlined className="site-form-item-icon" />}
                    onBlur={this.handlePassword}
                >
                </Input.Password>
                <p>请再次输入密码:</p>
                <Input.Password
                    placeholder="Enter your password again"
                    prefix={<UserOutlined className="site-form-item-icon" />}
                    onBlur={this.checkPassword}
                >
                </Input.Password>
                <div>
                    <p>性别：</p>
                    <Radio.Group onChange={this.handleGender} value={this.state.gender}>
                        <Radio value={0}>男</Radio>
                        <Radio value={1}>女</Radio>
                    </Radio.Group>
                </div>
                <div>
                    {this.state.passwordErrorVisible?(
                        <Alert message="请输入相同的密码" type="error"/>
                    ):null}
                </div>
                <p>请输入手机号:</p>
                <Input
                    placeholder="Enter your phone number"
                    prefix={<UserOutlined className="site-form-item-icon" />}
                    onBlur={this.handlePhone}
                >
                </Input>
                <p>请输入邮箱:</p>
                <Input
                    placeholder="Enter your address"
                    prefix={<UserOutlined className="site-form-item-icon" />}
                    onBlur={this.handleEmail}
                >
                </Input>
                <div>
                    {this.state.emailErrorVisible?(
                        <Alert message="请输入正确的Email" type="error"/>
                    ):null}
                </div>
                <p>上传头像：</p>
                <UploadImage onCommitImage={this.handleCommitImage}/>
                <Button onClick={this.handleSubmit.bind(this)}>
                    提交
                </Button>
                <div>
                    {(this.state.personicon==null)?null:(
                        <img className="personIcon" alt="personIcon" src={this.state.personicon}/>
                    )}
                </div>
            </div>
            </div>
        )
    }
}