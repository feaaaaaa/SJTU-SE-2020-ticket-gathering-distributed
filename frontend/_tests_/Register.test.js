jest.mock('node-fetch');
import React from 'react'
import {RegisterView} from "../src/view/RegisterView";
import {checkUser,register} from "../src/service/userService";
import fetch from 'node-fetch';

const {Response} = jest.requireActual('node-fetch');

describe("test register",()=>{
    const wrapper=shallow(<RegisterView/>);
    it("test initialized state",()=>{
        expect(wrapper.state().username).toBeNull();
        expect(wrapper.state().password).toBeNull();
        expect(wrapper.state().gender).toEqual(0);
        expect(wrapper.state().usernameErrorVisible).toBeFalsy();
        expect(wrapper.state().registered).toBeFalsy();
        expect(wrapper).toMatchSnapshot();
    });

    it("test input operation",()=>{
        wrapper.find('[placeholder="Enter your password"]').at(0).simulate('blur',{target:{value: '123456'}});
        expect(wrapper.state().password).toEqual('123456');
        wrapper.find('[placeholder="Enter your password again"]').at(0).simulate('blur',{target:{value: '12345'}});
        expect(wrapper.state().passwordErrorVisible).toBeTruthy();
        wrapper.find('[placeholder="Enter your email"]').at(0).simulate('blur',{target:{value: '123456'}});
        expect(wrapper.state().emailErrorVisible).toBeTruthy();
        wrapper.find('[placeholder="Enter your email"]').at(0).simulate('blur',{target:{value: 'oligei@gmail.com'}});
        expect(wrapper.state().emailErrorVisible).toBeFalsy();
        expect(wrapper.state().email).toEqual('oligei@gmail.com');
        // expect(wrapper).toMatchSnapshot();
    });

    it("test input username", async () => {
        fetch.mockReturnValue(Promise.resolve(new Response(true)));
        // await checkUser('oligei', (data) => {
        //     wrapper.setState({usernameErrorVisible: data})
        // });
        await wrapper.instance().handleUsername({target: {value: 'oligei'}});

        // new Promise(function(resolve){
        //     setTimeout(() => {
        //         wrapper.find('[placeholder="Enter your username"]').at(0).simulate('blur', {target: {value: 'oligei'}});
        //         resolve('success')
        //     }, 2000)
        // }).then(
        //     () => {
        //         expect(wrapper.state().usernameErrorVisible).toBeTruthy();
        //         console.log('test');
        //     },
        // ).catch(error=>{
        //     console.log(error);
        // });
        // await wrapper.find('[placeholder="Enter your username"]').at(0).simulate('blur', {target: {value: 'oligei'}});
        // console.log("test");
        expect(wrapper.state().usernameErrorVisible).toBeTruthy();
        // expect(wrapper).toMatchSnapshot();
    });

    it("test handle" , async ()=>{
        fetch.mockReturnValue(Promise.resolve(new Response(true)));
        await register("1","1","1","1","1",1,(data)=>{wrapper.setState({registerd:data})});
        expect(wrapper.state().registerd).toBeTruthy();
        // expect(wrapper).toMatchSnapshot();
    });
});




