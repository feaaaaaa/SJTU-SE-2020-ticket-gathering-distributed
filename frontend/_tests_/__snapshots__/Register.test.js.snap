// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test register test handle 1`] = `
<div
  id="bg"
>
  <div
    id="regDiv"
  >
    <p>
      请输入用户名:
    </p>
    <Input
      name="input"
      onBlur={[Function]}
      placeholder="Enter your username"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p>
      请输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <p>
      请再次输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password again"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <div />
    <div>
      <p>
        性别：
      </p>
      <Memo([object Object])
        onChange={[Function]}
        value={0}
      >
        <Radio
          type="radio"
          value="Male"
        >
          男
        </Radio>
        <Radio
          type="radio"
          value="Female"
        >
          女
        </Radio>
      </Memo([object Object])>
    </div>
    <p>
      请输入手机号:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your phone number"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <p>
      请输入邮箱:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your address"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p>
      上传头像：
    </p>
    <UploadImage
      onCommitImage={[Function]}
    />
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      提交
    </Button>
    <div />
  </div>
</div>
`;

exports[`test register test initialized state 1`] = `
<div
  id="bg"
>
  <div
    id="regDiv"
  >
    <p
      className="regp"
    >
      请输入用户名:
    </p>
    <Input
      className="input"
      onBlur={[Function]}
      placeholder="Enter your username"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p
      className="regp"
    >
      请输入密码:
    </p>
    <Password
      action="click"
      className="input"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <p
      className="regp"
    >
      请再次输入密码:
    </p>
    <Password
      action="click"
      className="input"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password again"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <div />
    <div>
      <p
        className="regp"
      >
        性别：
      </p>
      <Memo([object Object])
        onChange={[Function]}
        value={0}
      >
        <Radio
          style={
            Object {
              "color": "white",
            }
          }
          type="radio"
          value="Male"
        >
          男
        </Radio>
        <Radio
          style={
            Object {
              "color": "white",
            }
          }
          type="radio"
          value="Female"
        >
          女
        </Radio>
      </Memo([object Object])>
    </div>
    <p
      className="regp"
    >
      请输入手机号:
    </p>
    <Input
      className="input"
      onBlur={[Function]}
      placeholder="Enter your phone number"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <p
      className="regp"
    >
      请输入邮箱:
    </p>
    <Input
      className="input"
      onBlur={[Function]}
      placeholder="Enter your email"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p
      className="regp"
    >
      上传头像：
    </p>
    <UploadImage
      onCommitImage={[Function]}
    />
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      提交
    </Button>
    <div />
  </div>
</div>
`;

exports[`test register test input operation 1`] = `
<div
  id="bg"
>
  <div
    id="regDiv"
  >
    <p>
      请输入用户名:
    </p>
    <Input
      name="input"
      onBlur={[Function]}
      placeholder="Enter your username"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p>
      请输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <p>
      请再次输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password again"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <div>
      <Alert
        message="请输入相同的密码"
        type="error"
      />
    </div>
    <div>
      <p>
        性别：
      </p>
      <Memo([object Object])
        onChange={[Function]}
        value={0}
      >
        <Radio
          type="radio"
          value="Male"
        >
          男
        </Radio>
        <Radio
          type="radio"
          value="Female"
        >
          女
        </Radio>
      </Memo([object Object])>
    </div>
    <p>
      请输入手机号:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your phone number"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <p>
      请输入邮箱:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your address"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p>
      上传头像：
    </p>
    <UploadImage
      onCommitImage={[Function]}
    />
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      提交
    </Button>
    <div />
  </div>
</div>
`;

exports[`test register test input username 1`] = `
<div
  id="bg"
>
  <div
    id="regDiv"
  >
    <p>
      请输入用户名:
    </p>
    <Input
      name="input"
      onBlur={[Function]}
      placeholder="Enter your username"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div>
      <Alert
        message="该用户名已被注册过"
        type="error"
      />
    </div>
    <p>
      请输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <p>
      请再次输入密码:
    </p>
    <Password
      action="click"
      iconRender={[Function]}
      onBlur={[Function]}
      placeholder="Enter your password again"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      visibilityToggle={true}
    />
    <div />
    <div>
      <p>
        性别：
      </p>
      <Memo([object Object])
        onChange={[Function]}
        value={0}
      >
        <Radio
          type="radio"
          value="Male"
        >
          男
        </Radio>
        <Radio
          type="radio"
          value="Female"
        >
          女
        </Radio>
      </Memo([object Object])>
    </div>
    <p>
      请输入手机号:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your phone number"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <p>
      请输入邮箱:
    </p>
    <Input
      onBlur={[Function]}
      placeholder="Enter your address"
      prefix={
        <ForwardRef(UserOutlined)
          className="site-form-item-icon"
        />
      }
      type="text"
    />
    <div />
    <p>
      上传头像：
    </p>
    <UploadImage
      onCommitImage={[Function]}
    />
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      提交
    </Button>
    <div />
  </div>
</div>
`;
